### PR TITLE
Updates for cedar-policy/#714

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
*Issue #, if available:*

*Description of changes:*

### CODE REVIEW

File: CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java [hunk 1]
```diff
@@ -3,128 +3,64 @@
...
-public abstract class PartialAuthorizationResponse {
-    private final Diagnostics diagnostics;
+public class PartialAuthorizationResponse {
+    private Decision decision;
+    private Set<String> satisfied;
+    private Set<String> errored;
+    private Set<String> may_be_determining;
+    private Set<String> must_be_determining;
+    private Map<String, JsonNode> residuals;
+    private Set<String> nontrivial_residuals;
```
Comment (cdisselkoen) [inline CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java:23]: Should these be `final`?  I see that `diagnostics` was previously.  (Not a Java person and don't really know when we should/shouldn't use `final`)

File: CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java [hunk 2]
```diff
@@ -3,128 +3,64 @@
...
+        this.nontrivial_residuals = new HashSet<String>(nontrivial_residuals);
+    }
+
+    public Decision getDecision() {
+        return this.decision;
+    }
+
+    public Set<String> getResiduals() {
+            return this.residuals.keySet();
+    }
```
Comment (cdisselkoen) [inline CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java:42]: Should we have public getters for `errored`, `may_be_determining`, etc?

File: CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java [hunk 3]
```diff
@@ -3,128 +3,64 @@
...
+        this.errored = new HashSet<String>(errored);
+        this.may_be_determining = new HashSet<String>(may_be_determining);
+        this.must_be_determining = new HashSet<String>(must_be_determining);
+        this.residuals = new HashMap<String, JsonNode>(residuals);
+        this.nontrivial_residuals = new HashSet<String>(nontrivial_residuals);
+    }
+
+    public Decision getDecision() {
+        return this.decision;
+    }
```
Comment (cdisselkoen) [inline CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java:38]: should this have docs, eg, noting what returning `null` means?

**Review by cdisselkoen — APPROVED** [review]

File: CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java (hunk 2 above)
Comment (shaobo-he-aws) [reply CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java]: Done.

File: CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java (hunk 3 above)
Comment (shaobo-he-aws) [reply CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java]: Done.

File: CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java (hunk 1 above)
Comment (andrewmwells-amazon) [reply CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java]: IIUC `final` just means it won't change after construction, which seems appropriate here.

**Review by andrewmwells-amazon — APPROVED** [review]

**Pull request merged**